### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by mkaput**
---
This is **part 1 of 2 in a stack** made with GitButler:
- <kbd>&nbsp;2&nbsp;</kbd> #2 
- <kbd>&nbsp;1&nbsp;</kbd> #1 👈

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      non critical:
+        patterns:
+          - "*"
+        exclude-patterns:
+          - "cairo-lang-*"
+          - "lsp-types"
+          - "salsa"
+          - "scarb-*"
+    ignore:
+      - dependency-name: "cairo-lang-*"
```
Comment (piotmag769) [inline .github/dependabot.yml:22]: Not needed here, no?

**Review by piotmag769 — APPROVED** [review]

File: src/upgrade.rs [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    path: Option<PathBuf>,
+}
+
+pub fn main(args: Args) -> Result<()> {
+    let sh = Shell::new()?;
+
+    let mut cargo_toml = sh.read_file("Cargo.toml")?.parse::<DocumentMut>()?;
+
+    edit_dependencies(&mut cargo_toml, "dependencies", &args);
+    edit_dependencies(&mut cargo_toml, "dev-dependencies", &args);
```
Comment (maciektr) [inline src/upgrade.rs:59]: Does it work for workspaces without package in root? 🤔 (i.e. can this be defined when there is no package?)

**Review by maciektr — APPROVED** [review]

File: src/upgrade.rs (hunk 2 above)
Comment (mkaput) [reply src/upgrade.rs:59]: I don't expect such cases to happen anytime soon so I don't cover it

Comment (mkaput) [reply src/upgrade.rs:59]: Scarb has `[workspace.dependencies]` which is covered here as a third branch.

**Review by maciektr — APPROVED** [review]

**Pull request merged**